src/eureka/S4_generate_lightcurves/s4_genLC.py

```diff
@@ -223,44 +226,61 @@ def genlc(eventlabel, ecf_path=None, s3_meta=None):
+            
+            # Record and correct for 1D drift/jitter
+            if meta.recordDrift or meta.correctDrift:
                 # Calculate drift over all frames and non-destructive reads
                 # This can take a long time, so always print this message
-                log.writelog('Applying drift/jitter correction')
+                log.writelog('Recording drift/jitter')
                 # Compute drift/jitter
-                drift1d, driftmask = drift.spec1D(spec.optspec, meta, log,
-                                                  mask=spec.optmask)
+                drift_results = drift.spec1D(spec.optspec, meta, log,
+                                             mask=spec.optmask)
+                drift1d, driftwidth, driftmask = drift_results
                 # Replace masked points with moving mean
                 drift1d = clipping.replace_moving_mean(
                     drift1d, driftmask, Box1DKernel(meta.box_width))
-                lc['drift1d'] = (['time'], drift1d)
+                driftwidth = clipping.replace_moving_mean(
+                    driftwidth, driftmask, Box1DKernel(meta.box_width))
+                lc['driftxpos'] = (['time'], drift1d)
+                lc['driftxwidth'] = (['time'], driftwidth)
                 lc['driftmask'] = (['time'], driftmask)
-                # Correct for drift/jitter
-                for n in range(meta.n_int):
-                    # Need to zero-out the weights of masked data
-                    weights = (~spec.optmask[n]).astype(int)
-                    spline = spi.UnivariateSpline(np.arange(meta.subnx),
-                                                  spec.optspec[n], k=3, s=0,
-                                                  w=weights)
-                    spline2 = spi.UnivariateSpline(np.arange(meta.subnx),
-                                                   spec.opterr[n], k=3, s=0,
-                                                   w=weights)
-                    optmask = spec.optmask[n].astype(float)
-                    spline3 = spi.UnivariateSpline(np.arange(meta.subnx),
-                                                   optmask, k=3, s=0,
-                                                   w=weights)
-                    spec.optspec[n] = spline(np.arange(meta.subnx) +
-                                             lc.drift1d[n].values)
-                    spec.opterr[n] = spline2(np.arange(meta.subnx) +
-                                             lc.drift1d[n].values)
-                    # Also shift mask if moving by >= 0.5 pixels
-                    optmask = spline3(np.arange(meta.subnx) +
-                                      lc.drift1d[n].values)
-                    spec.optmask[n] = optmask >= 0.5
+                
+                spec['driftxpos'] = (['time'], drift1d)
+                spec['driftxwidth'] = (['time'], driftwidth)
+                spec['driftmask'] = (['time'], driftmask)
```

Comment:
I personally wouldn't bother putting these in spec - thoughts @kevin218?

Reply:
I would like to have them in spec, because people that do not want to use Eureka's S4 binned LC output and do their own binning will still want to have the x drifts recorded - for detrending purposes.